### PR TITLE
RESTWS-890: Support retrieving list of concepts by passing references

### PR DESCRIPTION
**Pull request by kdaud**
## Issue I worked on

see https://issues.openmrs.org/browse/RESTWS-890

## Checklist: I completed these to help reviewers :)

- [ ] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by kdaud** [comment]
> I'm curious about why you're targeting only 2.2+ for this PR?

I was thinking through this in the nutshell but could not exactly come up with the right way to go. I get to see the latest concept resource in version 2.1 and thought would have the service end point supported in 2.2+.

What would be the right way to implement the service end point? Is it to have it support from the least version(1.10+) or from the latest version(2.4+)?

**Comment by ibacher** [comment]
> What would be the right way to implement the service end point?

Generally having things able to run on the oldest version of OMRS possible is my preference.

**Comment by dkayiwa** [comment]
@kdaud isn't your new class simply duplicating what is already provided by this? https://github.com/openmrs/openmrs-module-webservices.rest/blob/master/omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/ConceptResource1_9.java

**Comment by kdaud** [comment]
> @kdaud isn't your new class simply duplicating what is already provided by this? https://github.com/openmrs/openmrs-module-webservices.rest/blob/master/omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/ConceptResource1_9.java

I am revising on this to have the endpoint implemented within 1.8

**Comment by dkayiwa** [comment]
This is the version of the OpenMRS platform required by this module: https://github.com/openmrs/openmrs-module-webservices.rest/blob/master/pom.xml#L36

**Comment by dkayiwa** [comment]
@kdaud what is your strategy for this. Is it as simple as boosting the exiting concept endpoint with a parameter that takes a comma separated list of UUIDs and SAME-AS mappings?

**Comment by kdaud** [comment]
> @kdaud what is your strategy for this

My initial plan was been creating a new concept endpoint. 

> Is it as simple as boosting the exiting concept endpoint with a parameter that takes a comma separated list of UUIDs and SAME-AS mappings?

The little/much I know is that the operation that need to be supported is the search operation when getting  a list of concepts. I am failing to identify which of the two methods is essential here. Is it the one in the [ConceptResource](https://github.com/openmrs/openmrs-module-webservices.rest/blob/42aa1d2f31283ea439c0cc887ba9f64f88f9a492/omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ConceptResource1_8.java#L542) or the one in [ConceptSearchHandler](https://github.com/openmrs/openmrs-module-webservices.rest/blob/42aa1d2f31283ea439c0cc887ba9f64f88f9a492/omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java#L70)?

**Comment by kdaud** [comment]
Or It's a different method that need to be boosted?

File: omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java [hunk 1]
```diff
@@ -53,8 +54,10 @@ public class ConceptSearchHandler1_8 implements SearchHandler {
 	            new SearchQuery.Builder("Allows you to find concepts by source and code").withRequiredParameters("source")
 	                    .withOptionalParameters("code").build(), new SearchQuery.Builder(
 	                    "Allows you to find concepts by name and class").withRequiredParameters("name")
-	                    .withOptionalParameters("class", "searchType").build()));
-	
+	                    .withOptionalParameters("class", "searchType").build(), 
+	                    new SearchQuery.Builder("Allows you to find concepts by reference")
+	                    .withRequiredParameters("reference").withOptionalParameters("uuid").build()));
+
```
Comment (kdaud) [inline omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:60]: I have added a new searchQuery which allows finding a concept by reference with optional parameter if uuid.

File: omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java [hunk 2]
```diff
@@ -53,8 +54,9 @@ public class ConceptSearchHandler1_8 implements SearchHandler {
 	            new SearchQuery.Builder("Allows you to find concepts by source and code").withRequiredParameters("source")
 	                    .withOptionalParameters("code").build(), new SearchQuery.Builder(
 	                    "Allows you to find concepts by name and class").withRequiredParameters("name")
-	                    .withOptionalParameters("class", "searchType").build()));
-	
+	                    .withOptionalParameters("class", "searchType").build(), 
+	                    new SearchQuery.Builder("Allows you to find concepts by uuid").withRequiredParameters("uuid").build()));
+
```
Comment (kdaud) [inline omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:59]: I have added a new searchQuery which allows finding a concept by uuid.

File: omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java [hunk 3]
```diff
@@ -73,9 +75,18 @@ public PageableResult search(RequestContext context) throws ResponseException {
 		String name = context.getParameter("name");
 		String conceptClass = context.getParameter("class");
 		String searchType = context.getParameter("searchType");
-		
+		String uuid = context.getParameter("uuid");
+		String reference = context.getParameter("reference");
+
 		List<Concept> concepts = new ArrayList<Concept>();
 		
+		ConceptSource cptSource = conceptService.getConceptByReference(reference);
```
Comment (kdaud) [inline omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:83]: @dkayiwa am getting an error when getting concept by reference and the log shows the method doesn't exit in core. Yet the method is [exists](https://github.com/openmrs/openmrs-core/blob/ffa3ebe927c18e29ec3c31627e050b99691b25fe/api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java#L345).

Is there anywhere am missing out? Kindly help

Comment (dkayiwa) [reply omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:83]: Did you understand this comment? https://github.com/openmrs/openmrs-module-webservices.rest/pull/556/#issuecomment-1162168774

Comment (kdaud) [reply omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:83]: > Did you understand this comment? #556

The link is taking me to the actual PR. Which comment is are you specifically referring to?

Comment (kdaud) [reply omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:83]: Are you probably referring to this [comment](https://github.com/openmrs/openmrs-module-webservices.rest/pull/556#issuecomment-1162893119)?

**Comment by dkayiwa** [comment]
> This is the version of the OpenMRS platform required by this module: https://github.com/openmrs/openmrs-module-webservices.rest/blob/master/pom.xml#L36

@kdaud this is the comment am referring to.

**Comment by kdaud** [comment]
I understood the comment as:

The webservices.rest module requires OpenMRS platform(openmrs-core) version 1.9.10 for it to compile successfully. In other-words the webservices.rest module depends on OpenMRS platform version 1.9.10. 

Could you help me understand the comment if my interpretation to the subject is not right?

**Comment by kdaud** [comment]
I also understand that OpenMRS platform is a term the community uses to refer to openmrs-core.

**Comment by dkayiwa** [comment]
The starting point to troubleshoot this error is to find out in which version of openmrs-core was this method introduced.

**Comment by kdaud** [comment]
The method was introduced in openmrs-core `since 2.6.0`

**Comment by dkayiwa** [comment]
Isn't this class already fetching concepts by uuid and mapping? https://github.com/openmrs/openmrs-module-webservices.rest/blob/master/omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/ConceptResource1_9.java

**Comment by dkayiwa** [comment]
Then what do you need getConceptByReference for?

**Comment by kdaud** [comment]
> Then what do you need getConceptByReference for?

Thanks @dkayiwa I get the point.

**Comment by kdaud** [comment]
@dkayiwa based on this [comment](https://github.com/openmrs/openmrs-module-webservices.rest/pull/556#issuecomment-1162168774) does it mean implementing the functionality in omod-1.9 module?

**Comment by dkayiwa** [comment]
Tru dat!

File: omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_9/ConceptSearchHandler1_9.java [hunk 4]
```diff
@@ -48,9 +50,9 @@ public class ConceptSearchHandler1_9 implements SearchHandler {
 	
 	private final SearchConfig searchConfig = new SearchConfig("byTerm", RestConstants.VERSION_1 + "/concept",
 			Collections.singletonList("1.9.* - 9.*"),
-	        new SearchQuery.Builder(
-	                "Allows you to find concepts which map to term, uuid of term given as input").withRequiredParameters(
-	            "term").build());
+			Arrays.asList(new SearchQuery.Builder("Allows you to find concepts which map to term, uuid of term given as input")
+					.withRequiredParameters("term").build(), new SearchQuery.Builder("Allows you to find concepts by uuid and mapping")
+					.withRequiredParameters("uuid").build()));
```
Comment (kdaud) [inline omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_9/ConceptSearchHandler1_9.java:55]: Hey @dkayiwa I have added a new search query in the existing endpoint. What have done is to introduce two search variables conceptUuid & mapping, and restricted them to be passed together when  requesting concepts. 

Is what have done ok?

File: omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_9/ConceptSearchHandler1_9.java [hunk 5]
```diff
@@ -48,9 +50,9 @@ public class ConceptSearchHandler1_9 implements SearchHandler {
 	
 	private final SearchConfig searchConfig = new SearchConfig("byTerm", RestConstants.VERSION_1 + "/concept",
 			Collections.singletonList("1.9.* - 9.*"),
-	        new SearchQuery.Builder(
-	                "Allows you to find concepts which map to term, uuid of term given as input").withRequiredParameters(
-	            "term").build());
+			Arrays.asList(new SearchQuery.Builder("Allows you to find concepts which map to term, uuid of term given as input")
+					.withRequiredParameters("term").build(), new SearchQuery.Builder("Allows you to find concepts by uuid and mapping")
+					.withRequiredParameters("uuid", "mapping").build()));
```
[the remark this answers is not in the document]
Comment (dkayiwa) [reply omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_9/ConceptSearchHandler1_9.java:55]: Before i respond with whether it is ok or not, does it work?

[the remark this answers is not in the document]
Comment (kdaud) [reply omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_9/ConceptSearchHandler1_9.java:55]: > does it work?

Not sure how I can test this to confirm? 

However,  theis functionality is to support the efficient loading of forms from the backend.

[the remark this answers is not in the document]
Comment (dkayiwa) [reply omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_9/ConceptSearchHandler1_9.java:55]: What does the JIRA ticket title say?

[the remark this answers is not in the document]
Comment (kdaud) [reply omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_9/ConceptSearchHandler1_9.java:55]: > What does the JIRA ticket title say?

Create a concept endpoint to support requesting multiple at once.

[the remark this answers is not in the document]
Comment (dkayiwa) [reply omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_9/ConceptSearchHandler1_9.java:55]: Does that work?

File: omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java [hunk 6]
```diff
@@ -73,8 +77,11 @@ public PageableResult search(RequestContext context) throws ResponseException {
 		String name = context.getParameter("name");
 		String conceptClass = context.getParameter("class");
 		String searchType = context.getParameter("searchType");
+		String uuid = context.getParameter("uuid");
+		String reference = context.getParameter("reference");
```
Comment (dkayiwa) [inline omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:81]: What is the use of the uuid and reference parameters?

Comment (kdaud) [reply omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:81]: Am using them to support retrieving concepts by both uuid and SAME-AS mapping.

Comment (kdaud) [reply omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:81]: Both of the parameters(uuid and reference) should be used when searching for concepts.

Comment (dkayiwa) [reply omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:81]: Isn't the current end point already fetching by both uuid and mapping?

Comment (kdaud) [reply omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:81]: I can only see fetching by [mapping ](https://github.com/openmrs/openmrs-module-webservices.rest/blob/42aa1d2f31283ea439c0cc887ba9f64f88f9a492/omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java#L142).

Comment (dkayiwa) [reply omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:81]: Did you test the end point and confirm that fetching by uuid does not work?

Comment (dkayiwa) [reply omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:81]: Have you actually tested the end point in your browser or any tools like curl, postman, etc and confirmed that the existing end point cannot fetch a concept by uuid and mapping?

Comment (kdaud) [reply omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:81]: > Have you actually tested the end point in your browser or any tools like curl, postman, etc and confirmed that the existing end point cannot fetch a concept by uuid and mapping?

I hadn't done this. Am doing it out to confirm preferably using postman

Comment (kdaud) [reply omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:81]: When you pass the uuid and mapping the endpoint can fetch(GET) a concept.

Comment (kdaud) [reply omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:81]: Thoughtfully the ticket  requirement is to support returning a list of(many) concepts from a single search.

Comment (kdaud) [reply omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:81]: > Then what did you mean by this? `What we do not have is a functionality that can fetch a concept by both uuid and mapping.`

I hadn't carried out the testing. So we do have the functionality.

File: omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java [hunk 7]
```diff
@@ -73,9 +76,28 @@ public PageableResult search(RequestContext context) throws ResponseException {
...
 		String conceptClass = context.getParameter("class");
 		String searchType = context.getParameter("searchType");
+		String conceptUuids = context.getParameter("uuids");
 		
 		List<Concept> concepts = new ArrayList<Concept>();
 		
+		if (StringUtils.isNotBlank(conceptUuids)) {
+			Concept concept = null;
+			for (String conceptUuid : conceptUuids.split(",")) {
+				if (!conceptUuid.trim().equalsIgnoreCase("")) {
```
Comment (ibacher) [inline omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:86]: `StringUtils.isNotBlank()` (from Apache Commons Lang)is a much more efficient and correct way to do this (basically, `trim()` will create a new String).

Comment (kdaud) [reply omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java]: > `StringUtils.isNotBlank()` (from Apache Commons Lang)is a much more efficient and correct way to do this (basically, `trim()` will create a new String).

I didn't know that trim() would rather create a new String.

**Comment by kdaud** [comment]
@dkayiwa @ibacher I am trying to use Postman to test out this commit. I am not aware where I can get the valid concept uuids for passing into the GET method.

**Comment by dkayiwa** [comment]
@kdaud are you simply asking for how to get a uuid of a concept in an OpenMRS database?

**Comment by dkayiwa** [comment]
@kdaud have you ever looked into an OpenMRS database?

**Comment by kdaud** [comment]
I have ever looked at https://wiki.openmrs.org/display/docs/Data+Model

**Comment by dkayiwa** [comment]
@kdaud and you still do not know how to get a uuid of a concept from the database?

**Comment by kdaud** [comment]
Have learnt it!

**Comment by ibacher** [comment]
@kdaud I hope you don't mind that I somewhat hijacked this PR... I still need to verify that this works.

**Comment by kdaud** [comment]
> I still need to verify that this works.

Will test it out over the night of tomorrow and will give the findings before Sunday morning!

**Comment by kdaud** [comment]
Have tested the functionality via postman and is working well. Any one else can test too and confirm.
cc: @ibacher

File: omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java [hunk 8]
```diff
@@ -53,7 +54,9 @@ public class ConceptSearchHandler1_8 implements SearchHandler {
 	            new SearchQuery.Builder("Allows you to find concepts by source and code").withRequiredParameters("source")
 	                    .withOptionalParameters("code").build(), new SearchQuery.Builder(
 	                    "Allows you to find concepts by name and class").withRequiredParameters("name")
-	                    .withOptionalParameters("class", "searchType").build()));
+	                    .withOptionalParameters("class", "searchType").build(), new SearchQuery.Builder(
+	                    		"Allows you to find a list of concepts by passing concept uuids")
+	                    .withRequiredParameters("references").build()));
```
Comment (dkayiwa) [inline omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:59]: Is `references` supposed to mean `uuid` and `mappings`?

Comment (dkayiwa) [reply omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:59]: Are we intentionally making `references` a required parameter?

**Comment by dkayiwa** [comment]
Do we intend to add some tests?

File: omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java (hunk 8 above)
Comment (kdaud) [reply omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:59]: > Is `references` supposed to mean `uuid` and `mappings`?

Actually `references` is meaning either `uuids` or `mapping`.

**Comment by kdaud** [comment]
> Do we intend to add some tests?

Have added some tests at https://github.com/kdaud/openmrs-module-webservices.rest/pull/3. Adding them here would override some work since my origin branch is missing changes. So was thinking to make a PR as soon as this is merged

**Comment by kdaud** [comment]
I am sorry @ibacher !

Instead of  updating this [PR](https://github.com/kdaud/openmrs-module-webservices.rest/pull/3) where I had written the unit tests, I mistakenly directed the changes at this commit and the work was overridden which I was avoiding at any cost. I was on a different branch RESTWS-890z and mistakenly directed the changes on this branch and I messed out.

File: omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java [hunk 9]
```diff
@@ -53,7 +55,9 @@ public class ConceptSearchHandler1_8 implements SearchHandler {
 	            new SearchQuery.Builder("Allows you to find concepts by source and code").withRequiredParameters("source")
 	                    .withOptionalParameters("code").build(), new SearchQuery.Builder(
 	                    "Allows you to find concepts by name and class").withRequiredParameters("name")
-	                    .withOptionalParameters("class", "searchType").build()));
+	                    .withOptionalParameters("class", "searchType").build(), new SearchQuery.Builder(
+	                    		"Allows you to find a list of concepts by passing concept uuids")
+	                    .withRequiredParameters("conceptUuids").build()));
```
Comment (dkayiwa) [inline omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:60]: Are we dealing with only concept uuids? How about mappings?

File: omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java (hunk 8 above)
Comment (ibacher) [reply omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:59]: > Is `references` supposed to mean uuid and mappings?

Technically, `references` is taken from the `getConceptByReference()` function for 2.6. `getConceptByReference()` has two other pathways that don't seem as relevant: one to load a constant from Java code via reflection with seems potentially dangerous to expose via REST and one to get concepts by the database id, which is generally not exposed through the REST API anyway. We could call it `mappingOrConceptUuid` but that seems a little verbose.

> Are we intentionally making references a required parameter?

Well, we've got three different supported queries here. The one that takes the `references` parameter doesn't accept any of the other parameters, so it's not exactly optional..

File: omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8Test.java [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+		return Context.getConceptService().getAllConcepts().size();
+	}
+
+	/**
+	 * @see ConceptSearchHandler1_8#search(GetConceptByReference)
+	 */
+	@Test
+	public void getConceptByReference_shouldReturnAListOfConceptsMatchingTheGivenReference() throws Exception {
+		final String searchString = "b9dbde07-2a0e-426e-b7d6-c8417fe82665";
+		SimpleObject result = deserialize(handle(newGetRequest(getURI(), new Parameter("q", searchString))));
```
Comment (ibacher) [inline omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8Test.java:52]: This should be:

```suggestion
		SimpleObject result = deserialize(handle(newGetRequest(getURI(), new Parameter("references", searchString))));
```

Right?

**Comment by kdaud** [comment]
I opted to add a new test file for conceptSearchHandler1_8 however, this is throwing aweird error: [UnknownResource](https://github.com/openmrs/openmrs-module-webservices.rest/runs/7282625943?check_suite_focus=true#step:6:1544). @ibacher should we instead add the tests in `ConceptSourceResource1_8Test` file?

File: omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java (hunk 8 above)
Comment (dkayiwa) [reply omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java:59]: Makes sense! 😊

**Comment by ibacher** [comment]
PS to updated to be able to see my changes locally, I think you just need to do a `git pull`.

**Comment by kdaud** [comment]
> Even though the omod-1.8 exists, it actually depends on OpenMRS 1.9 so the ConceptResource1_8 (which only supports 1.8.*) never loads.

It's weird !

**Comment by kdaud** [comment]
> PS to updated to be able to see my changes locally, I think you just need to do a `git pull`.

It works well!. Thanks

**Pull request merged**